core/unix/coredump.c

```diff
@@ -169,10 +170,18 @@ os_dump_core_internal(void)
     section_header_info_t section_header_info[MAX_SECTION_HEADERS + 1];
 
     memquery_iter_t iter;
-    if (!memquery_iterator_start(&iter, NULL, /*may_alloc=*/false)) {
+    if (!memquery_iterator_start(&iter, NULL, /*may_alloc=*/true)) {
         SYSLOG_INTERNAL_ERROR("memquery_iterator_start failed.");
         return false;
     }
+
+    ASSERT(d_r_get_num_threads() == 1);
+    // All other threads have been suspended, we can make it a shared table with
+    // lock.
```

Comment:
This seems backward: if we're the only thread, we could get away without the lock.  (Seems ok either way in the code, but the comment seems backward.)